### ISSUE TITLE
Float separators

### ISSUE THREAD

**Issue by Phyks** [spiral-project/ihatemoney#235]
Hi,

It seems the `input` field for the amount is of type `number`. Then, the browser enforces the use of the locale dependent comma separator. For instance, my French Firefox enforces the use of `,` as a float separator.

Problem is that on French keyboards, the `.` key in the numpad area types `.` and not `,`. Then, it is very unintuitive to type `3,2` instead of `3.2`.

Not sure if we should fix it on our side, but I was thinking some pre-processing JS code to convert `3.2` or `3,2` to the correct number representation for the browser could be useful.

EDIT: Just looking for feedbacks, I can contribute this if there is an agreement on this :)

**Comment by JocelynDelalande** [comment]
> Not sure if we should fix it on our side, but I was thinking some
> pre-processing JS code to convert |3.2| or |3,2| to the correct number
> representation for the browser could be useful.
>
Agreed. I'm hitting this too rigid behavior often also.

**Comment by aavenel** [comment]
In fact I think this is a regression, as I remember to write code to handle `,` or `.` without distinction in back-end validation. 

Perhaps using `<input type="number" formnovalidate />` will be enough to get rid of this. https://stackoverflow.com/a/15304154


```
class CommaDecimalField(DecimalField):
    """A class to deal with comma in Decimal Field"""
    def process_formdata(self, value):
        if value:
            value[0] = str(value[0]).replace(',', '.')
        return super(CommaDecimalField, self).process_formdata(value)
```

**Comment by Phyks** [comment]
Yes, either comma and dot are working in the backend code. But the browser does not validate float with dots with French locale. 

Not validating the field works, but "24,a34" will not be discarded either. I'll try to have a look at browsers capabilities. 
-- 
Phyks

**Comment by JocelynDelalande** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

Possible good compromise showed in :
https://codepen.io/anon/pen/MovRBW

It seems that you cannot customize validation as long as you keep
`type="number"`

-----BEGIN PGP SIGNATURE-----

iQIzBAEBCgAdFiEECtxJPuKxaTvavr2N6h2HoC+Us1MFAllOTdsACgkQ6h2HoC+U
s1OnOxAAgKFxn/ALVRwk1rsy8q9oHIMYM+5Gpw01XY42Rua4UfH2z8WYqmKBqKYE
+K/4CX6+fUHd6SFWxD8kChQ6cv336Ap/I+0u3rp405ugmfxCOkKcuStCtThjgOXQ
4Lob8kA8G0PMmnyAeK2Cq1CuxSrnLRUUwdq471zQbXwDXNI2/3+dnE5aKX0V5i1N
8Eplw/AcUS5yfBnYYHpr5yuPZbrXEuw7MkUI29kySAuYEDaPI6mYsg/48d/c9SMK
1MhXswZFK+jVBx02f6e9Wk7I66MS4afNPiteeUTKi5CKi4UbxuZG3E62Rm1pflYs
20w71oUPOUkkNGrTUOg41xu1Q0BWwj7jtGhwqb6dqBBLc8YB2X0Zy3kfDpAcSoIq
tk4b1qlpAANGVeOk2ciEqH2kDk4Dfoc8vh6PIMfjrsbAdegQ19MaKjUK4YgH0txh
sbYWFp4CxEoSZaFmAbyR6WOtJbkYwAgA4aUAeFhzJjm6FqmZMOiXiUwaTMxHgT0p
ikZFS9LqJxKgmUrfX3dHIdl3hzt5tu6lyIfUeOuWoYrqTS1KAC8Hy1DPCOuE7jhK
8rzZ+3ooFWNlVCQf1zY7wfAyaSJD2VDD66Lli+UfNVOOfmHteYaaEsy0YIlpahlg
cVdV8Pkb8qDuQSvVaKSqvJLFwsmaSWERhKyHdxTSeO8u//pFwVY=
=WoZP
-----END PGP SIGNATURE-----

**Comment by almet** [comment]
Yep, that looks like the way to go!

**Comment by JocelynDelalande** [comment]
Nice hack detailed here : https://www.ctrl.blog/entry/html5-input-number-localization

In short `<input type=number lang=en>` will accept both comma and dot as decimal separator.

**Comment by almet** [comment]
Let's just do it then?

**Comment by Phyks** [comment]
Not sure whether it's better to overload the `lang` or to use the ability to [pass a `regex`](https://stackoverflow.com/a/16883588). Maybe the latter is less hacky?

By the way, it seems this bug only affects Firefox, not Chromium, isn't it?

**Comment by JocelynDelalande** [comment]
> Not sure whether it's better to overload the lang or to use the ability to pass a regex. Maybe the latter is less hacky?

The first (`lang` stuff) is IMHO more user friendly, the user agent may tell « please enter a number » instead of « please enter a correct value ».

**Comment by Phyks** [comment]
I think this can be closed with #413 merged, isn't it?

**Comment by almet** [comment]
I believe, yes.
